src/core/unit.h

```diff
@@ -86,15 +86,19 @@ typedef enum UnitDependencyMask {
          * DefaultDependencies= and thus also involving configuration from UNIT_DEPENDENCY_FILE sources */
         UNIT_DEPENDENCY_MOUNTINFO_DEFAULT  = 1 << 6,
 
-        UNIT_DEPENDENCY_MASK_MOUNTINFO     = UNIT_DEPENDENCY_MOUNTINFO_IMPLICIT | UNIT_DEPENDENCY_MOUNTINFO_DEFAULT,
+        /* A dependency created because of data read from /proc/self/mountinfo, but fallback to unit configuration
+         * sources */
+        UNIT_DEPENDENCY_MOUNTINFO_OR_FILE  = 1 << 7,
```

Comment:
Can you explain the need for this new flag please, I'm not following. What problem is this commit supposed to solve exactly ?

Reply:
Yeah, this seems strange. This flag seems to be essentially saying "we don't know where this dependency came from exactly". It breaks the orthogonality of the different flags. After reading the commit message, I'd expect that we simply propagate the information about the appropriate mask from the caller down into the functions adding the dependencies. If they are added based on the unit file, use one mask; if they are added based on mountinfo, use the other mask.

Reply:
Strictly speaking, the dependencies tagged with this new label _always_ depend on /proc/self/mountinfo in some sense. Of course, there exists a relevant entry in /proc/self/mountinfo, then the dependencies depend on it. Even if there is no corresponding entry in /proc/self/mountinfo, the dependencies _are_ based on the fact that there is no entry in /proc/self/mountinfo. Hence, when /proc/self/mountinfo is updated, we need to regenerate the dependencies, and they must not be labeled with `UNIT_DEPENDENCY_FILE`.

Though, the name `UNIT_DEPENDENCY_MOUNTINFO_OR_FILE` may be misleading. But I have no idea about better name. Do you have any suggestions?